### PR TITLE
Add some basic ci and pin versions

### PR DESCRIPTION
**Pull request by jsturtevant**

### CODE REVIEW

**Review by danbugs — APPROVED** [review]
LGTM!

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+    name: Build
+    strategy:
+      matrix:
+        os: ["ubuntu-latest"]
+    runs-on: ${{ matrix.os }}
+    steps:
```
Comment (danbugs) [inline .github/workflows/ci.yaml:19]: ~might want to add Rust caching here to speed up things

Comment (jsturtevant) [reply .github/workflows/ci.yaml:19]: I believe this is setup automatically? https://github.com/actions-rust-lang/setup-rust-toolchain#inputs

<html>
<body>


cache | Automatically configure Rust cache (using Swatinem/rust-cache) | true
-- | -- | --



</body>
</html>

**Comment by jsturtevant** [comment]
it did take some time so hopefully the cache (which is now initialized) will help!

**Pull request merged**